http/src/main/scala/org/http4s/blaze/http/http2/Http2FrameEncoder.scala

```diff
@@ -0,0 +1,92 @@
+package org.http4s.blaze.http.http2
+
+import java.nio.ByteBuffer
+
+import org.http4s.blaze.util.BufferTools
+
+import scala.collection.immutable.VectorBuilder
+import scala.collection.mutable.ArrayBuffer
+
+/** A more humane interface for writing HTTP messages. */
+final class Http2FrameEncoder(
+    peerSettings: Http2Settings,
+    headerEncoder: HeaderEncoder) {
+
+  // Just a shortcut
+  private[this] def maxFrameSize: Int = peerSettings.maxFrameSize
+
+  /** Set the max table size of the header encoder */
+  def setMaxTableSize(size: Int): Unit =
+    headerEncoder.maxTableSize(size)
+
+  /** Generate a window update frame for the session flow window */
+  def sessionWindowUpdate(size: Int): ByteBuffer =
+    streamWindowUpdate(0, size)
+
+  /** Generate a window update frame for the specified stream flow window */
+  def streamWindowUpdate(streamId: Int, size: Int): ByteBuffer =
+    Http2FrameSerializer.mkWindowUpdateFrame(streamId, size)
+
+  /** Generate a ping frame */
+  def pingFrame(data: Array[Byte]): ByteBuffer =
+    Http2FrameSerializer.mkPingFrame(false, data)
+
+  /** Generate a ping ack frame with the specified data */
+  def pingAck(data: Array[Byte]): ByteBuffer =
+    Http2FrameSerializer.mkPingFrame(true, data)
+
+  /** Generate stream data frame(s) for the specified data
+    *
+    * If the data exceeds the peers MAX_FRAME_SIZE setting, it is fragmented
+    * into a series of frames.
+    */
+  def dataFrame(streamId: Int, endStream: Boolean, data: ByteBuffer): Seq[ByteBuffer] = {
+    val limit = maxFrameSize
+    if (data.remaining <= limit) Http2FrameSerializer.mkDataFrame(streamId, endStream, padding = 0, data)
+    else { // need to fragment
+      val acc = new ArrayBuffer[ByteBuffer]
```

Comment:
Why an ArrayBuffer here and a VectorBuilder in `headerFrame`?

Reply:
Code churn. I'll align them.